c2cgeoportal/views/entry.py

```diff
@@ -269,16 +280,20 @@ def _wms_getcap_cached(self, ogc_server, role_id):
             log.exception(error)
             return None, errors
 
+        # Fix for owslib / QGIS 3 compatibility
+        content = content.replace('<MetadataURL/>', '')
+
         try:
-            wms = WebMapService(None, xml=content)
-        except:  # pragma: no cover
+            wms = WebMapService(None, xml=content, version=version)
+        except Exception as e:  # pragma: no cover
             error = _(
                 "WARNING! an error occurred while trying to "
                 "read the mapfile and recover the themes."
             )
-            error = u"{0!s}\nURL: {1!s}\n{2!s}".format(error, url, content.encode("utf-8"))
+            error = u"{0!s}\nURL: {1!s}".format(error, url)
```

Comment:
Why removing the content?

Reply:
I removed content because it's hard to find error as content is very long.
Note that you have the url, easy to get content if needed.

Reply:
But many times the error is in the content, or related to the content...

Reply:
I note that it may be useful if status is 200 but response is a service exception.
In case response is a capabilities, like in my use case it's a PITA.
I did not plan to commit, I will revert this point.